src/norm/postgres.nim

```diff
@@ -403,7 +403,14 @@ proc selectOneToMany*[O: Model, M: Model](dbConn; oneEntries: seq[O], relatedEnt
 
   for entryId in entryIds:
     let id = entryId
-    relatedEntries[entryId] = relatedEntriesSeq.filter(target => target.dot(foreignKeyFieldName).id == id)
+    when M.dot(foreignKeyFieldName) is int64:
+      relatedEntries[entryId] = relatedEntriesSeq.filter(target => target.dot(foreignKeyFieldName) == id)
+    elif M.dot(foreignKeyFieldName) is Option[int64]:
+      relatedEntries[entryId] = relatedEntriesSeq.filter(target => target.dot(foreignKeyFieldName).get() == id)
+    elif M.dot(foreignKeyFieldName) is Option[O]:
+      relatedEntries[entryId] = relatedEntriesSeq.filter(target => target.dot(foreignKeyFieldName).get().id == id)
+    else:
+      relatedEntries[entryId] = relatedEntriesSeq.filter(target => target.dot(foreignKeyFieldName).id == id)
```

Comment:
Imo, postgres and sqlite can the same code in a seperate file. You can probably extract it into a templates.

Reply:
They have highly similar code, I'm struggling to split that code out properly. To just move that stuff into a template, the template would need to know about:

- M
- O
- foreignKeyFieldName
- relatedEntries
- rekatedEntriesSeq
- id/entryId

A template that takes in 6 parameters seems iffy to me.

Reply:
Can't we use filterIt instead of filter and avoid arrow proc syntax?

Reply:
@moigagoo : Done